src/sentry/static/sentry/app/utils/advancedAnalytics.tsx

```diff
@@ -0,0 +1,89 @@
+import * as qs from 'query-string';
+import {IntegrationsEventParams} from 'src/sentry/static/sentry/app/utils/integrationUtil';
+
+import {Organization} from 'app/types';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import {uniqueId} from 'app/utils/guid';
+
+const ANALYTICS_SESSION = 'ANALYTICS_SESSION' as const;
+
+export const startAnalyticsSession = () => {
+  const sessionId = uniqueId();
+  window.sessionStorage.setItem(ANALYTICS_SESSION, sessionId);
+  return sessionId;
+};
+
+export const clearAnalyticsSession = () => {
+  window.sessionStorage.removeItem(ANALYTICS_SESSION);
+};
+
+export const getAnalyticsSessionId = () =>
+  window.sessionStorage.getItem(ANALYTICS_SESSION);
+
+const hasAnalyticsDebug = () => window.localStorage.getItem('DEBUG_ANALYTICS') === '1';
+
+//TODO: add more types
+type EventParameters = IntegrationsEventParams;
```

Comment:
I wouldn't mind seeing this as something like

```tsx
const analyticKeys = {
	'thing.opened': 'Thing: Opened',
}
```

Then instead of having to actually specify the name, it can just be looked up from the key. Then we can type the key to `keyof analyticKeys`

Reply:
@EvanPurkhiser Yea but I want to enforce event schema depending on the event type. Like certain types of events should get other parameters. Unless you're talking about a way where we don't have to specify the `eventKey` and `eventName` separately and just look up the `eventName` based on the `eventKey`.

Reply:
that's fine we can do that as well with a type map

```tsx


type EventParameters = {
	'thing.opened': {
		thingId: string
	};
}

type AnalyticsKey = keyof EventParameters;

const analyticsKeyName: Record<AnalyticsKey, string> = {
	'thing.opened': 'Thing: Opened',
}

function sendAnAlytics<T extends AnalyticsKey>(
	key: T,
	params: RequriedParams & EventParameters[T]
) {
	...
}
```